LEAF_Request_Portal/templates/view_search.tpl

```diff
@@ -10,7 +10,38 @@ $(function() {
     var leafSearch = new LeafFormSearch('searchContainer');
     leafSearch.setOrgchartPath('<!--{$orgchartPath}-->');
 
+    var extendedQueryState = 0; // 0 = not run, 1 = need to process, 2 = processed
+    var foundOwnRequest = false;
+    var firstResult = {};
     query.onSuccess(function(res) {
+        // on the first run: if there are no results that are owned by the user,
+        // append requests owned by the user
+        if(extendedQueryState == 0) {
+            firstResult = res;
+            for(var i in res) {
+                if(res[i].userID == '<!--{$userID}-->') {
+                    foundOwnRequest = true;
+                    break;
+                }
+            }
+        }
+
+        if(extendedQueryState == 0
+            && foundOwnRequest == false
+            && leafSearch.getSearchInput() == '') {
+            extendedQueryState = 1;
+            query.addTerm('userID', '=', '<!--{$userID}-->');
+            query.execute();
+            return false;
+        }
+
+        if(extendedQueryState == 1) {
```

Comment:
I see what you mean here with the return statement. I believe since ```query``` is being reused to call ```.execute()``` then the onSuccess function will be called again. This is looking almost like a recursive query.

Reply:
Ah, I see now.  Thanks @thomasnguyen95

Reply:
Please approve if everything checks out